### PR TITLE
Feature/195 new _FLUSH events

### PR DESCRIPTION
**Pull request by kusmierz**
@TomHAnderson check this out (see #195 and f7965f8 comments).

### CODE REVIEW

File: src/Server/Resource/DoctrineResource.php [hunk 1]
```diff
@@ -348,6 +348,11 @@ public function create($data)
 
         $this->getObjectManager()->flush();
 
+        $results = $this->triggerDoctrineEvent(DoctrineResourceEvent::EVENT_CREATE_FLUSH, $entity, $data);
+        if ($results->last() instanceof ApiProblem) {
```
Comment (TomHAnderson) [inline src/Server/Resource/DoctrineResource.php]: This is confusing: if an API has already flushed then the data is persisted.  If we return an ApiProblem after persistence the client will not expect the data to have been persisted.  Perhaps _FLUSH will be unable to return an ApiProblem?

Comment (kusmierz) [inline src/Server/Resource/DoctrineResource.php]: I could change it, but I'm not sure why we should limit.

Secondly - could you see potential issue here? I think if somebody returns ApiProblem here, he knows what he is doing... For example you could create an entity (ie. photo) and next upload the image somewhere... and on error you could throw an ApiProblem // yeah, I know it's not the best idea, just an example.

Another thing - I'm trying to revert your change (using new event instead). There was the same `ApiProblem` logic.

Comment (TomHAnderson) [inline src/Server/Resource/DoctrineResource.php]: It's true events were not handled as they should and that's why they were reworked.

Will you modify the README.md and explain _FLUSH events with emphasis on why an ApiProblem could be returned even after the entity in question has been persisted?

FYI although I haven't used it I understand there's a file upload already in place in Apigility.

Comment (kusmierz) [inline src/Server/Resource/DoctrineResource.php]: OK, I've written in 26132ae631bfda1a8ec8b9c5217df4bae2d50b4c, but I'm not native english speaker, so could you check for language mistakes if any?

Comment (kusmierz) [inline src/Server/Resource/DoctrineResource.php]: @TomHAnderson Is it ok for you? Fix something?

**Comment by TomHAnderson** [comment]
Edited and included in #213

**Pull request closed without merging**